### ISSUE TITLE
Api realtime - Change Content-Type to text/event-stream for IIS reverse proxy compatibility

### ISSUE THREAD

**Issue by miro016** [pocketbase/pocketbase#3461]
Hi, great work with pocketbase. I'm attempting to host pocketbase behind an IIS reverse proxy. IIS, especially when coupled with the Application Request Routing (ARR) module, expects the Content-Type for Server-Sent Events (SSE) to be text/event-stream. However, the current configuration in pocketbase includes a charset parameter (text/event-stream; charset=UTF-8), which is causing compatibility issues.

Issue:
When pocketbase sends text/event-stream; charset=UTF-8 as the content type for realtime communication, IIS does not handle it correctly, preventing to establish connection.

Proposed Solution:
Please consider changing the Content-Type for SSE in pocketbase from text/event-stream; charset=UTF-8 to just text/event-stream. This would make it possible to host pocketbase behind an IIS reverse proxy without any issues. As far as I understand according to specification, charset is not required since utf-8 is default value.

**Comment by ganigeorgiev** [comment]
Could you provide some reference to an issue/doc that shows that the header value is the cause of the connection failure?

I don't exactly remember at the moment why I've added the charset, but eventually will consider removing for the upcoming v0.19.0 release.

**Comment by ganigeorgiev** [comment]
Btw. you can try override the header by attaching to the `OnRealtimeConnectRequest` hook. Here is an example using the JS `pb_hooks`:

```js
// pb_hooks/main.pb.js

onRealtimeConnectRequest((e) => {
    e.httpContext.response().header().set("Content-Type", "text/event-stream");
})
```

**Comment by miro016** [comment]
Unfortunately, There isn't widely circulated or recognized documentation (or I'm not able to find one) that speaks to specific issues or configurations of the content type in relation to IIS or its Application Request Routing (ARR) module. Only I found was one not related, outdated topic where is link to w3.org specification. 

However, maybe it will be useful for someone to write down how I proceeded :
- I've created nodejs app to simulate SSE communication according to this [tutorial](https://blog.q-bit.me/how-to-use-nodejs-for-server-sent-events-sse/). This was working for me even behind IIS reverse proxy
- Then I compared the response from pocketbase /api/realtime 
- The only difference was in content-type, so I changed header and it was working.

Link to [event-stream specification](https://html.spec.whatwg.org/multipage/iana.html#text/event-stream)

**Comment by miro016** [comment]
> Btw. you can try to override the header by attaching to the `OnRealtimeConnectRequest` hook. Here is an example using the JS `pb_hooks`:
> 
> ```js
> // pb_hooks/main.pb.js
> 
> onRealtimeConnectRequest((e) => {
>     e.httpContext.response().header().set("Content-Type", "text/event-stream");
> })
> ```

Yes, this is working for me. Thak you

**Comment by ganigeorgiev** [comment]
The charset was removed in the `develop` branch and the change will be available with the next v0.19.0 release in the near future.

I'm still not sure why it is causing compatibility issues in your case as specifying a `charset` parameter is valid `Content-type` header parameter, but I guess removing it will not break anything as the charset only denotes the messages encoding and as you've mentioned earlier, SSE spec complaint clients should expect it to be always UTF8 anyway (even in the [linked HTML Standard](https://html.spec.whatwg.org/multipage/iana.html#text/event-stream) is mentioned that the `charset` _"serves no purpose; it is only allowed for compatibility with legacy servers" so it should be safe in this case_).
